### DISCUSSION TITLE
Database notifications - ViewException thrown after installation

### DISCUSSION THREAD

**Discussion by studiowizjo** [filamentphp/filament#4009] [Help] [answered]
I would like to give a shot for the new feature - database notifications. Unfortunately after update, installation and configuration as described (both [in notifications docs](https://filamentphp.com/docs/2.x/notifications/database-notifications) and [admin panel docs](https://filamentphp.com/docs/2.x/admin/notifications#database-notifications)) I got an error:

```php
Filament\Notifications\Http\Livewire\Notifications::getDatabaseNotificationsQuery(): Return value must be of type Illuminate\Contracts\Database\Eloquent\Builder, Illuminate\Database\Eloquent\Relations\MorphMany returned (View: C:\wamp64\www\filament2\vendor\filament\notifications\resources\views\components\database\index.blade.php)

public function getDatabaseNotificationsQuery(): Builder
{
    /** @phpstan-ignore-next-line */

    return $this->getUser()->notifications()->where('data->format', 'filament');
}
```

 I am using admin panel, notifications package and MySQL database.

I did also one additional thing which is not mentioned in docs, but I think is necessary:

```php
use Illuminate\Notifications\Notifiable;

class User extends Authenticatable
{
    use Notifiable;

    //...
}
```

**Comment by danharrin** [comment, marked as answer]
Hey! Are you on Laravel 8 or 9?

**Reply by danharrin** [reply]
Probably fixed by #4013

**Reply by studiowizjo** [reply]
Indeed, it works! Thanks :).

BTW. It was Laravel  8.83.23
